src/Server/ValidationRules/RequireAuthentication.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace WPGraphQL\Server\ValidationRules;
+
+use GraphQL\Error\Error;
+use GraphQL\Language\AST\FieldNode;
+use GraphQL\Language\AST\NodeKind;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Validator\Rules\QuerySecurityRule;
+use GraphQL\Validator\ValidationContext;
+use WPGraphQL\AppContext;
+
+/**
+ * Class RequireAuthentication
+ *
+ * @package WPGraphQL\Server\ValidationRules
+ */
+class RequireAuthentication extends QuerySecurityRule {
+
+	/**
+	 * @return bool
+	 */
+	protected function isEnabled() {
+
+		$restrict_endpoint = null;
+
+		/**
+		 * Allows overriding the default graphql_restrict_endpoint behavior. Returning anything other
+		 * than null will skip the default restrict checks.
+		 *
+		 * @param null            $restrict_endpoint null
+		*/
+		$restrict_endpoint = apply_filters( 'graphql_pre_restrict_endpoint', $restrict_endpoint );
+
+		if ( null !== $restrict_endpoint ) {
+			return (bool) $restrict_endpoint;
+		}
+
+		// Check to see if the endpoint should be restricted to logged in users
+		$restrict_endpoint = get_graphql_setting( 'restrict_endpoint_to_logged_in_users' );
+
+		if ( false === is_graphql_http_request() ) {
+			return false;
+		}
+
+		if ( empty( $restrict_endpoint ) ) {
+			return false;
+		}
+
+		if ( 'on' !== $restrict_endpoint ) {
+			return false;
+		}
+
+		if ( null !== wp_get_current_user() && 0 !== wp_get_current_user()->ID ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * @param ValidationContext $context
+	 *
+	 * @return callable[]|mixed[]
+	 */
+	public function getVisitor( ValidationContext $context ) {
```

Comment:
Where is this getVisitor() used? Am I missing something?

Reply:
@markkelnar the `getVisitor` method comes from the ValidationRule class in GraphQL-PHP: https://github.com/webonyx/graphql-php/blob/0d43d7bee3fdcaf99537f6265203b7692f39e6ca/src/Validator/Rules/ValidationRule.php#L33-L43